cmd/influxd/run/server_test.go

```diff
@@ -994,6 +994,66 @@ func TestServer_Query_Count(t *testing.T) {
 	}
 }
 
+// Ensure that aggregate queries work properly after a WAL flush
+func TestServer_QueryAfterFlush(t *testing.T) {
+	t.Parallel()
+	s := OpenServer(NewConfig(), "")
+	defer s.Close()
+
+	if err := s.CreateDatabaseAndRetentionPolicy("db0", newRetentionPolicyInfo("rp0", 1, 1*time.Hour)); err != nil {
+		t.Fatal(err)
+	}
+
+	test := NewTest("db0", "rp0")
+	writes := []string{
+		"cpu,host=server1 value=1.0 10000000000",
+		"cpu,host=server1 value=2.0 20000000000",
+		"cpu,host=server1 value=3.0 30000000000",
+	}
+	test.write = strings.Join(writes, "\n")
+
+	test.addQueries([]*Query{
+		&Query{
+			name:    "selecting count(value) should succeed",
```

Comment:
Maybe add a second test here that does the `count` without any `GROUP BY`, and enhance names of tests to call out difference.